rmgpy/solver/base.pyx

```diff
@@ -915,67 +841,139 @@ cdef class ReactionSystem(DASx):
+                        interrupt = True
+                
+                sortedInds = numpy.argsort(numpy.array(tempNewObjectVals)).tolist()[::-1]
+                
+                newObjects.extend([tempNewObjects[q] for q in sortedInds])
+                newObjectInds.extend([tempNewObjectInds[q] for q in sortedInds])
+                newObjectVals.extend([tempNewObjectVals[q] for q in sortedInds])
+                
+                tempNewObjects = []
+                tempNewObjectInds = []
+                tempNewObjectVals = []
+                
+            #remove excess objects
+            if len(invalidObjects) + len(newObjects) > maxNumObjsPerIter:
+                logging.info('Exceeded max number of objects...removing excess objects')
+                num = maxNumObjsPerIter - len(invalidObjects)
+                newObjects = newObjects[:num]
+                newObjectInds = newObjectInds[:num]
+                newObjectVals = newObjectVals[:num]
+                
+            if terminateAtMaxObjects and len(invalidObjects)+len(newObjects) >= maxNumObjsPerIter:
+                logging.info('Reached max number of objects...preparing to terminate')
+                interrupt = True
+            
+            schanged = False
+            
+            if newObjects != []:
+                for i,obj in enumerate(newObjects): #log everything and add reactions to surface
+                    val = newObjectVals[i]
+                    ind = newObjectInds[i]
+                    if isinstance(obj,Species):
+                        logging.info('At time {0:10.4e} s, species {1} at rate ratio {2} exceeded the minimum rate for moving to model core of {3}'.format(self.t, obj,val,toleranceMoveToCore))
+                    elif isinstance(obj,Reaction):
+                        if i in newSurfaceRxnInds:
+                            logging.info('At time {0:10.4e} s, Reaction {1} at {2} exceeded the minimum difference in total log(accumulation number) for moving to model surface of {3}'.format(self.t, obj, val,toleranceMoveEdgeReactionToSurface))
+                            newSurfaceReactions.append(obj)
+                            newSurfaceReactionInds.append(ind)
+                        else:
+                            logging.info('At time {0:10.4e} s, Reaction {1} at {2} exceeded the minimum difference in total log(accumulation number) for moving to model core of {3}'.format(self.t, obj, val,toleranceMoveEdgeReactionToCore))
+                    else: 
+                        logging.info('At time {0:10.4e} s, PDepNetwork #{1:d} at {2} exceeded the minimum rate for exploring of {3}'.format(self.t, obj.index, val,toleranceMoveToCore))
+    
+                
+                invalidObjects += newObjects
+                
+            if schanged: #reinitialize surface
```

Comment:
Good catch!  I must've gotten confused and been thinking I needed to track if something got added to the surface, but that doesn't matter since it won't even be in the core until the simulation finishes.  It shouldn't be set to false that way.